sleap/io/visuals.py

```diff
@@ -112,34 +112,38 @@ def writer(
     total_frames_written = 0
     start_time = perf_counter()
     i = 0
-    while True:
-        data = in_q.get()
-
-        if data is _sentinel:
-            # no more data to be received so stop
-            in_q.put(_sentinel)
-            break
+    try:
+        while True:
+            data = in_q.get()
 
-        if writer_object is None and data:
-            h, w = data[0].shape[:2]
-            writer_object = VideoWriter.safe_builder(
-                filename, height=h, width=w, fps=fps
-            )
+            if data is _sentinel:
+                # no more data to be received so stop
+                in_q.put(_sentinel)
+                break
 
-        t0 = perf_counter()
-        for img in data:
-            writer_object.add_frame(img, bgr=True)
+            if writer_object is None and data:
+                h, w = data[0].shape[:2]
+                writer_object = VideoWriter.safe_builder(
+                    filename, height=h, width=w, fps=fps
+                )
 
-        elapsed = perf_counter() - t0
-        fps = len(data) / elapsed
-        logger.debug(f"writing chunk {i} in {elapsed} s = {fps} fps")
-        i += 1
+            t0 = perf_counter()
+            for img in data:
+                writer_object.add_frame(img, bgr=True)
 
-        total_frames_written += len(data)
-        total_elapsed = perf_counter() - start_time
-        progress_queue.put((total_frames_written, total_elapsed))
+            elapsed = perf_counter() - t0
+            fps = len(data) / elapsed
+            logger.debug(f"writing chunk {i} in {elapsed} s = {fps} fps")
+            i += 1
 
-    writer_object.close()
+            total_frames_written += len(data)
+            total_elapsed = perf_counter() - start_time
+            progress_queue.put((total_frames_written, total_elapsed))
+    except Exception as e:
+        raise (e)
+    finally:
+        if writer_object is not None:
+            writer_object.close()
     # send (-1, time) to signal done
     progress_queue.put((-1, total_elapsed))
```

Comment:
We are still able to get stuck in the code somewhere even with this `try except finally` executing as expected. For example, I intentionally errored-out to this message but was not returned to the CLI:
```
(sleap_render-video-crop) λ python tests\liezl_test_script.py
Saving config: C:\Users\TalmoLab/.sleap/1.2.4/preferences.yaml
Writing video with 14100 frame images...
INFO:sleap.io.visuals:Chunks: 221, chunk size: 64
Done in 0.41996690000000036 s, fps = 33574.07452825446.
Exception in thread Thread-3:
Traceback (most recent call last):
  File "C:\Users\TalmoLab\miniconda3\envs\sleap_render-video-crop\lib\threading.py", line 926, in _bootstrap_inner
    self.run()
  File "C:\Users\TalmoLab\miniconda3\envs\sleap_render-video-crop\lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "d:\social-leap-estimates-animal-poses\pull-requests\sleap_render-video-crop\sleap\sleap\io\visuals.py", line 145, in writer
    raise (e)
  File "d:\social-leap-estimates-animal-poses\pull-requests\sleap_render-video-crop\sleap\sleap\io\visuals.py", line 126, in writer
    h, w = data[0].shape[:2]
AttributeError: 'NoneType' object has no attribute 'shape'
```

Reply:
Opting to push the crop video fixes in this PR and handle the threading hang-up as a different issue.